vent/controller/control_module.py

```diff
@@ -319,148 +234,98 @@ def __analyze_last_waveform(self):
                 self._DATA_I_PHASE = np.nan
 
             # and measure the breaths per minute
-            self._DATA_BPM = 60. / phase[-1]  # 60 sec divided by the duration of last waveform
-
-    def __update_alarms(self):
-        ''' This goes through the values obtained from the last waveform, and updates alarms.'''
-        if len(self.__cycle_waveform_archive) > 1 : # Only if there was a previous cycle
-            self.__test_critical_levels(min=self.__PIP_min, max=self.__PIP_max, value=self._DATA_PIP, name=ValueName.PIP)
-            self.__test_critical_levels(min=self.__PIP_time_min, max=self.__PIP_time_max, value=self._DATA_PIP_TIME, name=ValueName.PIP_TIME)
-            self.__test_critical_levels(min=self.__PEEP_min, max=self.__PEEP_max, value=self._DATA_PEEP, name=ValueName.PEEP)
-            self.__test_critical_levels(min=self.__bpm_min, max=self.__bpm_max, value=self._DATA_BPM, name=ValueName.BREATHS_PER_MINUTE)
-            self.__test_critical_levels(min=self.__I_phase_min, max=self.__I_phase_max, value=self._DATA_I_PHASE, name=ValueName.INSPIRATION_TIME_SEC)
+            try:
+                self._DATA_BPM = 60. / phase[-1]  # 60 sec divided by the duration of last waveform, exception if this was 0.
+            except:
```

Comment:
when does this happen? seems like a pretty broad catch to me

Reply:
One of the things @gdoubleyew brought up. Want to make sure that the code doesn't crash, so I'm catching division by zero, nans etc. Nothing of this _should_ happen, but better be safe

Reply:
ok gotcha.